### PR TITLE
Default granularity bug fix: handle time dimensions with the same name

### PR DESCRIPTION
**Pull request by courtneyholcomb**
### Description
Found another bug in SetDefaultGranularityRule. If there are multiple `agg_time_dimensions` with the same name across semantic models, and they have different granularities, we'll only check one. Fixed here!

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: dbt_semantic_interfaces/transformations/default_granularity.py [hunk 1]
```diff
@@ -34,7 +38,7 @@ def transform_model(semantic_manifest: PydanticSemanticManifest) -> PydanticSema
                 continue
 
             default_granularity = TimeGranularity.DAY
-            seen_agg_time_dimensions: Set[TimeDimensionReference] = set()
+            seen_agg_time_dimensions: Set[Tuple[SemanticModel, TimeDimensionReference]] = set()
```
Comment (WilliamDee) [inline dbt_semantic_interfaces/transformations/default_granularity.py:41]: Im always wary of using larger objects in sets, so if we could change this to like a semanticmodelreference would be nice

**Review by WilliamDee — APPROVED** [review]

File: dbt_semantic_interfaces/transformations/default_granularity.py (hunk 1 above)
Comment (courtneyholcomb) [reply dbt_semantic_interfaces/transformations/default_granularity.py:41]: Good call

File: dbt_semantic_interfaces/transformations/default_granularity.py [hunk 2]
```diff
@@ -49,11 +53,16 @@ def transform_model(semantic_manifest: PydanticSemanticManifest) -> PydanticSema
                     except AssertionError:
                         # This indicates the agg_time_dimension is misconfigured, which will fail elsewhere.
                         # Do nothing here to avoid disrupting the validation process.
+                        logger.warning(
+                            f"Agg time dimension '{agg_time_dimension_ref.element_name}' not found in model."
+                            "This should raise a validation error elsewhere."
```
Comment (plypaul) [inline dbt_semantic_interfaces/transformations/default_granularity.py:58]: Where is it supposed to raise a validation error?

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "dbt-semantic-interfaces"
-version = "0.6.3"
+version = "0.6.4"
```
Comment (plypaul) [inline pyproject.toml:3]: Can you move this to a separate PR?

**Review by plypaul — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (courtneyholcomb) [reply pyproject.toml:3]: Shoot, sorry got Will's approval and merged before I got this 😅

File: dbt_semantic_interfaces/transformations/default_granularity.py (hunk 2 above)
Comment (courtneyholcomb) [reply dbt_semantic_interfaces/transformations/default_granularity.py:58]: https://github.com/dbt-labs/dbt-semantic-interfaces/blob/49a0347597b60fbec8d17771fa561c858f06b046/dbt_semantic_interfaces/validations/agg_time_dimension.py#L47-L48
